docs/tools/bambi_appsscript.md

```diff
@@ -0,0 +1,109 @@
+[**UDMI**](../../) / [**Docs**](../) / [**Tools**](./) / [BAMBI Sheet Interface Setup](#)
+
+# Getting Started with the BAMBI Sheet Interface
+
+This guide provides instructions on how to set up a Google Sheet to communicate with the BAMBI backend service.
+
+## Overview
+
+The setup involves two Google Apps Scripts that work together:
+
+1.  **BAMBISheetsInterface**: This script is bound directly to your Google Sheet. It creates the necessary tabs, formatting, and custom menu options you'll use to interact with BAMBI.
+2.  **PubSubPublisherWebApp**: This is a standalone "web app" script that publishes messages to the BAMBI backend using Google's Pub/Sub service.
+
+We use two separate scripts because a script must be linked to a Google Cloud Platform (GCP) project to get permission to publish messages. By having one central `PubSubPublisherWebApp`, multiple users can create their own `BAMBISheetsInterface` sheets without needing complex GCP configurations.
+
+## Creating Your BAMBI Sheet
+
+Follow these instructions to create your own copy of the BAMBI sheet. 
+
+### 🚀 Automated Setup
+
+This method uses a Python script to create and configure your new BAMBI sheet automatically.
+
+
+#### **One-Time Prerequisites**
+
+Before running the script for the first time, you must enable the Apps Script API.
+If you do not want to enable the Apps Script API in your project, you can jump to the Manual Setup.
+
+1.  **Enable the API in your GCP Project**: Visit the following URL and enable the API. Replace `${GCP_PROJECT_ID}` with your actual Google Cloud Project ID.
+    * `https://console.developers.google.com/apis/api/script.googleapis.com/overview?project=${GCP_PROJECT_ID}`
+2.  **Enable the API for your Google Account**: Visit your Apps Script settings and turn on the "Google Apps Script API."
+    * `https://script.google.com/home/usersettings`
+
+#### **Instructions**
+
+1.  **Authenticate with Google Cloud**: Run the following commands in your terminal to log in and set the correct permissions.
+
+    ```shell
+    gcloud auth login
+    gcloud config set project ${GCP_PROJECT_ID}
+    gcloud auth application-default login \
+    --scopes="openid,[https://www.googleapis.com/auth/userinfo.email,https://www.googleapis.com/auth/cloud-platform,https://www.googleapis.com/auth/spreadsheets,https://www.googleapis.com/auth/script.projects,https://www.googleapis.com/auth/drive.metadata.readonly](https://www.googleapis.com/auth/userinfo.email,https://www.googleapis.com/auth/cloud-platform,https://www.googleapis.com/auth/spreadsheets,https://www.googleapis.com/auth/script.projects,https://www.googleapis.com/auth/drive.metadata.readonly)"
```

Comment:
This line shouldn't have the MD in it!  E.g. I deleted the []s and everything between the ()s and it worked better.